lib/dice/src/mfg.rs

```diff
@@ -0,0 +1,272 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    cert::{Cert, DeviceIdSelfCertBuilder},
+    csr::DeviceIdCsrBuilder,
+    CertSerialNumber,
+};
+use dice_mfg_msgs::{MfgMessage, SerialNumber, SizedBlob};
+use lib_lpc55_usart::{Read, Usart, Write};
+use nb;
+use salty::signature::Keypair;
+
+pub enum Error {
+    MsgDecode,
+    MsgBufFull,
+    UsartRead,
+    UsartWrite,
+}
+
+// data returned to caller by MFG
+// typically required to use ECA post MFG
+// this is what gets written to persistent storage after successful mfg
```

Comment:
Can you define ECA?

Reply:
Apologies, this is a DICE-ism. ECA is "embedded certificate authority" and in this case it's the DeviceId cert. These comments probably should have been removed before the PR & they'll be replaced by code in my next PR.